gnucash/register/register-gnome/completioncell-gnome.c

```diff
@@ -0,0 +1,1010 @@
+        return;
+
+    box = cell->cell.gui_private;
+    if (box == NULL)
+        return;
+
+    if (box->item_list != NULL)
+    {
+        g_hash_table_remove_all (box->item_hash);
+        item_store_clear (cell);
+    }
+}
+
+void
+gnc_completion_cell_add_menu_item (CompletionCell* cell,
+                                   const char* menustr)
+{
+    PopBox* box;
+
+    if (cell == NULL)
+        return;
+    if (menustr == NULL)
+        return;
+
+    box = cell->cell.gui_private;
+
+    if (box->item_list != NULL)
+    {
+        if (box->register_is_reversed)
+        {
+            box->occurrence_id++;
+            if (g_hash_table_lookup_extended (box->item_hash, menustr, NULL, NULL))
+                return;
+        }
+        else
+            box->occurrence_id--;
+
+        g_hash_table_insert (box->item_hash, g_strdup (menustr), GINT_TO_POINTER(box->occurrence_id));
+    }
+}
+
+void
+gnc_completion_cell_set_value (CompletionCell* cell, const char* str)
+{
+    if (cell == NULL)
+        return;
+    if (str == NULL)
+        return;
+
+    gnc_basic_cell_set_value (&cell->cell, str);
+}
+
+static inline void
+list_store_append (GtkListStore *store, char* string, char* string_norm, gint occurrence_id)
+{
+    GtkTreeIter iter;
+
+    g_return_if_fail (store != NULL);
+    g_return_if_fail (string != NULL);
+    g_return_if_fail (string_norm != NULL);
+
+    gtk_list_store_append (store, &iter);
+    gtk_list_store_set (store, &iter, TEXT_COL, string,
+                                      TEXT_NORMAL_COL, string_norm,
+                                      OCCURANCE_COL, occurrence_id, -1);
+}
+
+static char*
+casefold (char* utf8_string)
+{
+    char *folded = NULL;
+    g_return_val_if_fail (utf8_string && *utf8_string, NULL);
+
+    folded = g_utf8_casefold (utf8_string, -1);
+    return folded;
+}
+
+static char*
+normalize (char* utf8_string)
+{
+    char *normalized = NULL;
+    g_return_val_if_fail (utf8_string && *utf8_string, NULL);
+
+    normalized = g_utf8_normalize (utf8_string, -1, G_NORMALIZE_NFC);
+    return normalized;
+}
+
+static char*
+normalize_and_fold (char* utf8_string)
+{
+    char *normalized, *folded;
+    g_return_val_if_fail (utf8_string && *utf8_string, NULL);
+
+    normalized = normalize (utf8_string);
+    if (!normalized)
+        return NULL;
+
+    folded = casefold (normalized);
+    g_free (normalized);
+    return folded;
+}
+
+static void
+add_item_anywhere (gpointer key, gpointer value, gpointer user_data)
+{
+    PopBox* box = user_data;
+    gchar *hash_entry = g_strdup (key);
+
+    box->using_anywhere = TRUE;
+
+    if (hash_entry && *hash_entry)
+    {
+        gnc_utf8_strip_invalid_and_controls (hash_entry);
+        gchar *key_norm      = normalize (hash_entry);
+        gchar *key_norm_fold = casefold (key_norm);
+
+        if (g_strstr_len (key_norm_fold, -1, box->newval) != NULL)
+            list_store_append (box->item_store, hash_entry, key_norm, GPOINTER_TO_INT(value));
```

Comment:
I couldn't determine where you are using the second column of the list store (the one in which you store `key_norm`). What am I missing ?

Reply:
That was to do with sorting alphabetically but that was scrapped so removed.